src/models/yamlProjectConfiguration.ts

```diff
@@ -318,146 +285,171 @@ export class YamlProjectConfiguration {
+                }
+
+                return true;
+            }),
+            region: zod.enum(regions).default("us-east-1"),
+            language: zod.nativeEnum(Language).default(Language.ts),
+            cloudProvider: zod
+                .nativeEnum(CloudProviderIdentifier)
+                .default(CloudProviderIdentifier.GENEZIO),
+            classes: zod
+                .array(
+                    zod
+                        .object({
+                            path: zod.string(),
+                            type: zod.nativeEnum(TriggerType).default(TriggerType.jsonrpc),
+                            name: zod.string().optional(),
+                            methods: zod.array(methodConfigurationSchema).optional(),
+                        })
+                        // Hack to make sure that the method type is set to the class type
+                        .transform((value) => {
+                            for (const method of value.methods || []) {
+                                method.type = method.type || value.type;
+                            }
+
+                            return value;
+                        }),
+                )
+                .optional(),
+            options: zod.object({ nodeRuntime: zod.enum(supportedNodeRuntimes) }).optional(),
+            sdk: zod
+                .object({
+                    language: zod.nativeEnum(Language),
+                    path: zod.string(),
+                })
+                .optional(),
+            frontend: zod
+                .object({
+                    path: zod.string(),
+                    subdomain: zod.string().optional(),
+                })
+                .optional(),
+            workspace: zod
+                .object({
+                    backend: zod.string(),
+                    frontend: zod.string(),
+                })
+                .optional(),
+            packageManager: zod.nativeEnum(PackageManagerType).optional(),
+            scripts: zod
+                .object({
+                    preBackendDeploy: zod.string().optional(),
+                    postBackendDeploy: zod.string().optional(),
+                    postFrontendDeploy: zod.string().optional(),
+                    preFrontendDeploy: zod.string().optional(),
+                    preStartLocal: zod.string().optional(),
+                    postStartLocal: zod.string().optional(),
+                    preReloadLocal: zod.string().optional(),
+                })
+                .optional(),
+            plugins: zod
+                .object({
+                    astGenerator: zod.array(zod.string()),
+                    sdkGenerator: zod.array(zod.string()),
+                })
+                .optional(),
+            publicSdk: zod
+                .object({
+                    name: zod.string(),
+                    version: zod.string(),
+                    public: zod.boolean().default(true),
+                })
+                .optional(),
```

Comment:
The public sdk was reverted this morning. This shouldn't belong here probably. 

Rebase to the dev branch.